server/server.go

```diff
@@ -200,6 +203,9 @@ func (s *server) Run(rootCtx context.Context) (err error) {
 	if s.network != nil {
 		stopCtx, stopCtxCancel := context.WithTimeout(context.Background(), StopOnSignalTimeout)
 		defer stopCtxCancel()
+		// we need to cancel the Start context so that any pending action with that context
+		// can be cancelled
+		s.cancelFunc()
```

Comment:
we need also to add this cancelation to the Stop() function of the server, so CLI stop command also cancels a probably concurrently running Start(), without needing to shutdown the server